lib/CL/pocl_binary.c

```diff
@@ -128,11 +128,14 @@ typedef struct pocl_binary_s
 /***********************************************************/
 
 #define BUFFER_STORE(elem, type)                  \
-  *(type*)buffer = (type)TO_LE((type)elem);       \
-  buffer += sizeof(type)
+  do {                                            \
+    type b_s_tmp = (type)TO_LE((type)elem);       \
```

Comment:
Otherwise looks good but please mind the style (GNU C style in C files). https://www.gnu.org/prep/standards/html_node/Formatting.html#Formatting  at least the function argument lists need additional white spaces. Like: func_call (arguments, here);  Yes, the original was also wrongly formatted, but the intention is to get every future commit to improve on this matter also.